File: python/trees/0094-binary-tree-inorder-traversal.py
from typing import List
from structures import TreeNode


def inorderTraversal(root: TreeNode) -> List[int]:
    if root is None:
        return []
    
    output = []
    if root.left:
        output.extend(inorderTraversal(root.left))
        
    output.append(root.val)
    
    if root.right:
        output.extend(inorderTraversal(root.right))
        
    return output

File: js/trees/0094-binary-tree-inorder-traversal.js
var inorderTraversal = function(root) {
    let result = [];
    dfs(root);
    
    function dfs(root) {
        if(root !== null) {
            dfs(root.left);
            result.push(root.val);
            dfs(root.right);
        }
    }
    
    return result;
};